src/diff/props.js

```diff
@@ -11,12 +11,17 @@ import options from '../options';
  */
 export function diffProps(dom, newProps, oldProps, isSvg) {
 	for (let i in newProps) {
-		if (i!=='children' && i!=='key' && (!oldProps || oldProps[i]!=newProps[i])) {
+		if (i!=='children' && i!=='key' && i!=='value' && (!oldProps || oldProps[i]!=newProps[i])) {
 			setProperty(dom, i, newProps[i], oldProps[i], isSvg);
 		}
 	}
+
+	// `value` needs to be applied after an input's `min`, `max` and `step`
+	// attribute are set. See preact/#850
+	if (newProps.value!=null) dom.value = newProps.value;
+
 	for (let i in oldProps) {
-		if (i!=='children' && i!=='key' && (!newProps || !(i in newProps))) {
+		if (i!=='children' && i!=='key' && !(i in newProps)) {
```

Comment:
Is the removal of `!newProps` check because `newProps` is guaranteed to be non-null here?

Reply:
yup, we always set at least an empty `object` https://github.com/developit/preact/blob/0baf368ca803e0deb6b3d7edd2e09f6bbd564505/src/create-element.js#L12